src/main/java/com/hrznstudio/titanium/material/ResourceMaterial.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of Titanium
+ * Copyright (C) 2019, Horizon Studio <contact@hrznstudio.com>.
+ *
+ * This code is licensed under GNU Lesser General Public License v3.0, the full license text can be found in LICENSE.txt
+ */
+
+package com.hrznstudio.titanium.material;
+
+import com.hrznstudio.titanium.api.material.IResourceType;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.registries.ForgeRegistryEntry;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ResourceMaterial {
+
+    private String type;
+    private HashMap<String, IResourceType> generatorTypes;
```

Comment:
WHy not just a Table or a HashMap<String<Pair<IResourceType, ResourceTypeProperties>>